src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -0,0 +1,233 @@
+		frontLeftMotor = new CANSparkMax(HardwareMap.CAN_ID_SPARK_DRIVE_FRONT_LEFT, CANSparkMax.MotorType.kBrushless);
+		backRightMotor = new CANSparkMax(HardwareMap.CAN_ID_SPARK_DRIVE_BACK_RIGHT, CANSparkMax.MotorType.kBrushless);
+		backLeftMotor = new CANSparkMax(HardwareMap.CAN_ID_SPARK_DRIVE_BACK_LEFT, CANSparkMax.MotorType.kBrushless);
+
+		
+		frontRightMotor.getEncoder().setPosition(0);
+		frontLeftMotor.getEncoder().setPosition(0);
+		backRightMotor.getEncoder().setPosition(0);
+		backLeftMotor.getEncoder().setPosition(0);
+		// Reset state machine
+		reset();
+	}
+
+	/* ======================== Public methods ======================== */
+	/**
+	 * Return current FSM state.
+	 * @return Current FSM state
+	 */
+	public FSMState getCurrentState() {
+		return currentState;
+	}
+	/**
+	 * Reset this system to its start state. This may be called from mode init
+	 * when the robot is enabled.
+	 *
+	 * Note this is distinct from the one-time initialization in the constructor
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = FSMState.TELEOP_STATE;
+
+		// Call one tick of update to ensure outputs reflect start state
+		update(null);
+	}
+	/**
+	 * Update FSM based on new inputs. This function only calls the FSM state
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+		switch (currentState) {
+			case START_STATE:
+				handleStartState(input);
+				break;
+
+			case TELEOP_STATE:
+				handleTeleOpState(input);
+                break;
+                
+			case FORWARD_STATE_10_IN:
+				handleForwardOrBackwardState(input, 10);
+				break;
+
+			case TURN_STATE:
+				handleTurnState(input, 90); // test 90 degrees
+				break;
+
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+		currentState = nextState(input);
+	}
+
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private FSMState nextState(TeleopInput input) {
+		switch (currentState) {
+			case START_STATE:
+				if (input != null) {
+					return FSMState.TELEOP_STATE;
+				} else {
+					return FSMState.START_STATE;
+				}
+
+			case TELEOP_STATE:
+				return FSMState.TELEOP_STATE;
+            
+			case FORWARD_STATE_10_IN:
+				return FSMState.TURN_STATE;
+				//should be return FSMState.TURN_RIGHT; once simon pushes his code
```

Comment:
Again, I assume there's some condition check missing here. You have to wait for the robot to be done moving forward before starting to turn right?